test/runtests.jl

```diff
@@ -1,9 +1,22 @@
 using TulipaProfileFitting
 using Test
 
-@testset "TulipaProfileFitting.jl" begin
+function test_fit_profile()
+
+    # Test case 1: Using root finding method
     profile_values = [0.2, 0.5, 0.9, 1.0, 0.4, 0.1, 0.0]
     target_mean = 0.5
-    coefficient = fit_profile(profile_values, target_mean)
-    @test coefficient ≈ 0.7491921789832345 atol = 0.01
+    expected_coefficient = 0.7491921789832345
+    @test fit_profile(profile_values, target_mean) ≈ expected_coefficient atol = 1e-3
+
+    # Test case 2: Using optimization method
+    target_mean = 1.0
+    expected_coefficient = 0.0
+    @test fit_profile(profile_values, target_mean) ≈ expected_coefficient atol = 1e-3
```

Comment:
I understand what you're doing, and the code seems correct, but I don't know what you want to do, so I am not sure it this is the best approach.
This solution is to change all profile values to 1. Is that what you want in this situation?
Are the profiles always between 0.0 and 1.0? Is that why you use p^x?

Reply:
The test is fine. I am just not sure if this is what you want for your scaler because to me it doesn't make sense to "scale" all values to 1.0.

Reply:
I've updated the tests to make them more meaningful in our expected results. If the input data is not in per unit ([0,1]) then you need either one method (`find_zero`) or the other (`optimization`) depending on the target value. You can double-check in the new test the values that I proposed. Nevertheless, it is fair to say that most of our profiles will be per unit, which means we will mainly use the `find_zero` part.